src/settings/accounts-settings.tsx

```diff
@@ -104,174 +95,126 @@ const getAvailableEmailAddresses = (account: Account, settings: AccountSettings)
 	return uniq(result);
 };
 
+type IdentityAttrsRecord<T extends string | number> = Record<T, Partial<IdentityAttrs>>;
+
 export const AccountsSettings = ({
 	account,
 	identitiesDefault,
 	t
 }: AccountSettingsProps): ReactElement => {
-	const [mods, setMods] = useState<Mods>({});
-	const [activeDelegateView, setActiveDelegateView] = useState('0');
+	const [createList, setCreateList] = useState<NonNullable<IdentityAttrsRecord<number>>>({});
+	const [deleteList, setDeleteList] = useState<Array<string>>([]);
+	const [modifyList, setModifyList] = useState<NonNullable<IdentityAttrsRecord<string>>>({});
+
+	const resetLists = useCallback(() => {
+		setCreateList({});
+		setDeleteList([]);
+		setModifyList({});
+	}, []);
+
 	const [selectedIdentityId, setSelectedIdentityId] = useState(0);
 	const [identities, setIdentities] = useState<IdentityProps[]>(identitiesDefault);
-	const [delegates, setDelegates] = useState<DelegateType[]>([]);
 	const settings = useUserSettings();
 	const maxIdentities = settings.attrs.zimbraIdentityMaxNumEntries;
-	const addMod = useCallback(
-		(arg: AddModProps) => {
-			const { type, modifyProp, deleteList, createList } = arg;
-			setMods((prevState) => {
-				const prevRecord = find(
-					prevState?.identity?.modifyList,
-					(item) => item.id === modifyProp?.id
-				)?.prefs;
-				const modifyList =
-					typeof modifyProp !== 'undefined'
-						? {
-								...prevState.identity?.modifyList,
-								[modifyProp.id]: {
-									id: modifyProp.id,
-									prefs: { ...prevRecord, [modifyProp.key]: modifyProp.value }
-								}
-						  }
-						: prevState.identity?.modifyList;
-				const newCreateList = prevState.identity?.createList || createList;
-				const newDeleteList = prevState.identity?.deleteList || deleteList;
-				return {
-					...prevState,
 
-					[type]: {
-						deleteList: newDeleteList,
-						createList: newCreateList,
-						modifyList
-					}
-				};
-			});
-		},
-		[setMods]
-	);
+	const onCancel = useCallback(() => {
+		resetLists();
+		setIdentities(identitiesDefault);
+	}, [identitiesDefault, resetLists]);
+
+	const updateModifyList = useCallback<
+		<K extends keyof IdentityAttrs>(id: string, key: K, value: IdentityAttrs[K]) => void
+	>(
+		(id, key, value) => {
+			const updatedIdentityKey = lowerFirst(replace(key, 'zimbraPref', '')) as keyof IdentityProps;
 
-	const modifyCreateList = useCallback((arg: AddModProps) => {
-		const { type, modifyProp } = arg;
-		setMods((prevState) => {
-			const prevRecord = find(
-				prevState?.identity?.createList,
-				(item) => item.prefs.requestId === modifyProp?.id
-			)?.prefs;
-			const modifiedCreateList =
-				typeof modifyProp !== 'undefined'
-					? {
-							...prevState.identity?.createList,
-							[modifyProp.id]: {
-								prefs: { ...prevRecord, [modifyProp.key]: modifyProp.value }
-							}
-					  }
-					: undefined;
-			return {
-				...prevState,
+			const prevRecordPrefs = modifyList[id] || {};
```

Comment:
remove the dependency with the state modifyList by using the prevState of the setModifyList